packages/migrate/src/__tests__/__snapshots__/DbPull.test.ts.snap

```diff
@@ -146,6 +181,45 @@ enum Role {
 }
 
 
+// introspectionSchemaVersion: NonPrisma,
+`;
+
+exports[`cockroachdb basic introspection (with postgresql schema) 2`] = `
+generator client {
+  provider = "prisma-client-js"
+}
+
+datasource db {
+  provider = "postgresql"
+  url      = env("TEST_COCKROACH_URI")
+}
+
+model Post {
+  id        String    @id
+  createdAt DateTime  @default(now())
+  updatedAt DateTime  @default(dbgenerated("'1970-01-01 00:00:00':::TIMESTAMP"))
+  published Boolean   @default(false)
+  title     String
+  content   String?
+  authorId  String?
+  jsonData  Json?
+  coinflips Boolean[]
+  User      User?     @relation(fields: [authorId], references: [id])
+}
+
+model User {
+  id    String  @id
+  email String  @unique(map: "User.email")
+  name  String?
+  Post  Post[]
+}
```

Comment:
We should probably make some of the `String` fields different native types, that are different between CockroachDB and PostgreSQL provider.

Reply:
So, in this case they should be `VarChars`, right? Since we have a postgresql schema

Reply:
On a closer inspection, the way these models are generated is not dependent on the code shipped with this PR, as both the postgres schema and the cockroach one have no model defined. We are merely introspecting existing tables in the postgres and cockroach images running on Docker. I have just merged `main` in `integration/fix-db-pull-url-overwrites-provider-cockroachdb` to trigger the CI, let's see if anything changes.

Reply:
The CI succeeds nonetheless, I suggest we open a new issue to verify why the introspection results in "general types" rather than "native types" (if that's unexpected at all), and merge this PR in the meantime.
Do you agree? @janpio @Jolg42